### ISSUE TITLE
Enviar as novas mensagens de um chamado de forma que elas se agrupem

### ISSUE THREAD

**Issue by VirusABC** [uspdev/chamados#279]
Olá

A cada chamado, modificação e atualização nos chamados, o sistema envia um e-mail.

Seria legal se esse e-mail viesse de forma que os diversos clientes de e-mail consigam agrupar essas mensagens em uma mesma thread para facilitar acompanhar os chamados via e-mail. No momento, o Gmail agrupa de alguma forma, mas o meu cliente de e-mail não agrupa, provavelmente porque O Message-ID das mensagens nos Headers do e-mail é diferente a cada mensagem e o sistema não coloca um "Re: " na frente do assunto para cada mensagem subsequente. Seria legal se o Message-ID fosse mantido e as novas postagens viessem com um "Re: "

No meu caso, utilizo claws-mail que é um cliente de e-mail simples.

**Comment by masakik** [comment]
Ola, o message-id é unico para cada email. Provavelmente o gmail esteja agrupando pois o subject é identico para todas as mensagens relacionadas.

**Comment by VirusABC** [comment]
Verdade, o Message-ID é único... mas e-mails respondidos podem ter um
trecho "References" com ids de mensagens passadas e um trecho
"In-Reply-To" com a message-id da mensagem que foi respondida.

Acho que se juntar um "Re: " antes do assunto para as próximas
mensagens vindas de um mesmo chamado, já ajudaria muito outros clientes
de e-mail. 


Em Thu, 06 May 2021 06:23:56 -0700
Masaki Kawabata Neto ***@***.***> escreveu:

> Ola, o message-id é unico para cada email. Provavelmente o gmail
> esteja agrupando pois o subject é identico para todas as mensagens
> relacionadas.
> 



-- 
Andre Lopes da Silva
Técnico em Informática
SI - IME-USP
3091-9891
Sala 120-A

Enviado utilizando Claws Mail
OpenSUSE Leap

**Comment by masakik** [comment]
Acho que pode dar certo. Você consegue mandar um PR para testarmos?

**Comment by VirusABC** [comment]
Desculpe, o que seria um PR?


Em Thu, 06 May 2021 06:53:14 -0700
Masaki Kawabata Neto ***@***.***> escreveu:

> Acho que pode dar certo. Você consegue mandar um PR para testarmos?
> 
> 



-- 
Andre Lopes da Silva
Técnico em Informática
SI - IME-USP
3091-9891
Sala 120-A

Enviado utilizando Claws Mail
OpenSUSE Leap

**Comment by masakik** [comment]
PR = Pull Request. Seria uma das formas de contribuir com o projeto. Você codificaria essa feature e mandaria o código para ser juntado ao codigo principal por meio de um pull request.
Acho que vc não mexe com git certo? Programa em PHP, saberia fazer essas alterações?
Se o problema for git podemos marcar um meet sobre isso. Agora se vc não programa em PHP então nem adianta. 
Eu não vou conseguir mexer por agora pois estou trabalhando em outro projeto. Quando eu liberar podemos ver isso. Também precisaria de sua ajuda pois não mexo com esse cliente de email e não teria onde testar.
Você está usando em produção ou pensa em usar na sua unidade?

**Comment by masakik** [comment]
Assim que eu liberar um pouco vejo algumas possibilidades e conversamos. Hoje coincidentemente, em um chamado que participo o gmail não conseguiu agrupar a última mensagem que recebi e agrupou outras 5 mensagens anteriores. :smiley:

**Comment by VirusABC** [comment]
Acho que o Gmail deve agrupar coisas com o mesmo assunto que chegam em
um certo espaço de tempo... depois ele não agrupa mais... Por exemplo,
quando recebo uma transferência bancária, o banco sempre me manda um
aviso com o mesmo título... Quando recebo duas transferências em um
menor espaço de tempo, o gmail agrupa... mas se for uma em um dia e
outra em outro, aí já fica fora.

Na verdade, pensando agora, faz sentido ele não agrupar para sempre...
imagina toda vez ressuscitar uma mensagem só porque foi enviada uma
mensagem com mesmo assunto. Depois de um tempo, a chance de o assunto
ser apenas uma coincidência pode ser maior do que ser uma "resposta" ao
mesmo e-mail.

Abraços

Em Tue, 11 May 2021 16:31:45 -0700
Masaki Kawabata Neto ***@***.***> escreveu:

> Assim que eu liberar um pouco vejo algumas possibilidades e
> conversamos. Hoje coincidentemente, em um chamado que participo o
> gmail não conseguiu agrupar a última mensagem que recebi e agrupou
> outras 5 mensagens anteriores. :smiley: 
> 



-- 
Andre Lopes da Silva
Técnico em Informática
SI - IME-USP
3091-9891
Sala 120-A

Enviado utilizando Claws Mail
OpenSUSE Tumbleweed
